blockchain/validate.go

```diff
@@ -675,11 +675,26 @@ func (b *BlockChain) checkBlockHeaderContext(header *wire.BlockHeader, prevNode
 		blockDifficulty := header.Bits
 		if blockDifficulty != expDiff {
 			str := fmt.Sprintf("block difficulty of %d is not the"+
-				" expected "+"value of %d", blockDifficulty,
+				" expected value of %d", blockDifficulty,
 				expDiff)
 			return ruleError(ErrUnexpectedDifficulty, str)
 		}
 
+		// Ensure the stake difficulty specified in the block header
+		// matches the calculated difficulty based on the previous block
+		// and difficulty retarget rules.
+		expSDiff, err := b.calcNextRequiredStakeDifficulty(prevNode)
+		if err != nil {
+			log.Errorf("calcNextRequiredStakeDifficulty: %v", err)
+			return err
```

Comment:
why is the error both logged and returned, probably to be logged again?

Reply:
The error here would only be a generic failure to load a block node from the database which doesn't have any context associated with it, so the extra log here help identify the specific failure path should it be hit.

Reply:
can more context be added to the error, instead of handling it twice?

Reply:
I just nuked the log message.  This is all going to change in future commits anyways such that dynamic block loading will go away.